test/configuration_unittest.cc

```diff
@@ -0,0 +1,61 @@
+#include "../src/configuration.h"
+#include "gtest/gtest.h"
+
+namespace {
+
+void TestDefaultConfig(google::MetadataAgentConfiguration& config) {
+  EXPECT_EQ("", config.ProjectId());
+  EXPECT_EQ("", config.CredentialsFile());
+  EXPECT_EQ(3, config.MetadataApiNumThreads());
+  EXPECT_EQ(8000, config.MetadataApiPort());
+  EXPECT_EQ(".", config.MetadataApiResourceTypeSeparator());
+  EXPECT_EQ(60, config.MetadataReporterIntervalSeconds());
+  EXPECT_EQ(false, config.MetadataReporterPurgeDeleted());
+  EXPECT_EQ(
+      "https://stackdriver.googleapis.com/"
+      "v1beta2/projects/{{project_id}}/resourceMetadata:batchUpdate",
```

Comment:
We should eventually add a `CheckConfiguration` function that verifies, e.g., that these format strings have the appropriate substitution parameters, etc. Not asking you to do it now, though.

Reply:
Agreed, left for later